app/androidutils.cpp

```diff
@@ -35,3 +35,24 @@ bool AndroidUtils::isAndroid() const
     return false;
 #endif
 }
+
+void AndroidUtils::requirePermissions()
+{
+    checkAndAcquirePermissions("android.permission.WRITE_EXTERNAL_STORAGE");
```

Comment:
on my Android 6 phone things worked without being asked for this permission... is it possible to do Android version check and only ask on Android >= 8 ?

Reply:
For my Android 6 and 7, it has to be there otherwise data folder is not accessible. Have you properly uninstall the app? There was no such issue with SDK 21 though. However, a new problem has been introduce on startup. If permission are accepted, on the first lunch of app there are missing demo projects and even all downloaded projects are tagged as invalid. From second and further lunch it is all ok.